experiments/experiment_data.py

```diff
@@ -125,6 +128,24 @@ def _extract_perf_config_key(self, perf_config):
         return self._make_pa_key_from_cli_string(pa_string)
 
     def _make_pa_key_from_cli_string(self, pa_cli_string):
-        concurrencies = re.findall('--concurrency-range=\d+', pa_cli_string)
-        key = ';'.join(concurrencies)
-        return key
+        concurrencies = re.findall('--concurrency-range=(\d+)', pa_cli_string)
+        batch_sizes = re.findall(' -b (\d+)', pa_cli_string)
+
+        if len(concurrencies) != len(batch_sizes):
+            raise Exception(f"concurrencies don't match batch sizes")
+
+        for i in range(len(concurrencies)):
+            tmp_int = int(concurrencies[i]) * int(batch_sizes[i])
+            clamped_int = self._clamp_to_power_of_two(tmp_int)
+            concurrencies[i] = str(clamped_int)
+
+        return ';'.join(concurrencies)
```

Comment:
Should we attach the `--concurrency-range=` prefix back before we return?

Reply:
It doesn't matter. All reads (getting experiment data) and writes (loading checkpoint data into this class) go through this function.